torch/csrc/jit/codegen/cuda/kernel_db/kernel_db.cpp

```diff
@@ -0,0 +1,253 @@
+#include <mutex>
+#include <regex>
+
+#include <torch/csrc/jit/codegen/cuda/instrumentation.h>
+#include "torch/csrc/jit/codegen/cuda/kernel_db/kernel_db.h"
+#include "torch/csrc/jit/codegen/cuda/kernel_db/utils.h"
+#include "torch/csrc/jit/codegen/cuda/utils.h"
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+static std::mutex kernel_db_lock;
+
+KernelDb::KernelDb(bool _disabled)
+    : disabled_(_disabled),
+      initialized_(false),
+      kernel_map_(),
+      kernel_db_path_(),
+      kernel_db_txt_file_() {}
+
+bool KernelDb::open(
+    const std::string& kernel_db_dir,
+    const std::string& kernel_db_file,
+    bool use_temp_dir) {
+  FUSER_PERF_SCOPE("KernelDb::open");
+  const std::string header(
```

Comment:
Nit: Move definition below creating/querying the DB.

Reply:
Done.